examples/timeseries/event_classification_for_payment_card_fraud_detection.py

```diff
@@ -0,0 +1,432 @@
+"""
+Title: Event classification for payment card fraud detection
+Author: [achoum](https://github.com/achoum/)
+Date created: 2024/02/01
+Last modified: 2024/02/01
+Description: Detection of fraudulent payment card transactions using Temporian and a feed-forward neural network.
+Accelerator: GPU
+"""
+
+"""
+This notebook depends on Keras 3, Temporian, and a few other libraries. You can install them as follow:
+
+```shell
+pip install temporian keras pandas tf-nightly scikit-learn -U
+```
+"""
+
+import keras  # To train the Machine Learning model
+import temporian as tp  # To convert transactions into tabular data
+
+import numpy as np
+import os
+import pandas as pd
+import datetime
+import math
+import tensorflow as tf
+from sklearn.metrics import RocCurveDisplay
+
+"""
+## Introduction
+
+Payment fraud detection is critical for banks, businesses, and consumers. In Europe alone, fraudulent transactions were estimated at [€1.89 billion in 2019](https://www.ecb.europa.eu/pub/pdf/cardfraud/ecb.cardfraudreport202110~cac4c418e8.en.pdf). Worldwide, approximately [3.6%](https://www.cybersource.com/content/dam/documents/campaign/fraud-report/global-fraud-report-2022.pdf) of commerce revenue is lost to fraud. In this notebook, we train and evaluate a model to detect fraudulent transactions using the synthetic dataset attached to the book [Reproducible Machine Learning for Credit Card Fraud Detection](https://fraud-detection-handbook.github.io/fraud-detection-handbook/Foreword.html) by Le Borgne et al.
```

Comment:
Please keep lines under 80 chars, this facilitates code review and version control.

Reply:
Good point.
*Done